<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="plot" default="false"/>
    <arg name="enable_obstacle" default="true"/>
    <arg name="holonomic" default="true" />
    <arg name="debug" default="false"/>

    <arg name="experiment" default="false"/>
    <arg name="experiment_id" default="manipulation_target_reaching"/>
    <!-- TF prefix -->
    <arg name="tf_prefix" default=""/>
    
    <!--  Run the test node-->
    <arg name="launch_prefix" value="gdb -ex run --args" if="$(arg debug)"/>
    <arg name="launch_prefix" value="" unless="$(arg debug)"/>

    <include file="$(find husky_panda_manipulation)/launch/load.launch">
        <arg name="holonomic" value="$(arg holonomic)" />
    </include>
    
    <!-- Load robot description (for simulation) -->
    <!-- <arg name="robot_description_file" default="$(find husky_panda_manipulation)/data/husky_panda_planar.urdf" if="$(arg holonomic)"/> -->
    <arg name="robot_description_file" default="$(find husky_panda_manipulation)/data/ridgeback_panda.urdf" if="$(arg holonomic)"/>
    <arg name="robot_description_file" default="$(find husky_panda_manipulation)/data/husky_panda.urdf" unless="$(arg holonomic)"/>

    <!-- Publish robot state (joints) as TF -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
        <param name="publish_frequency" value="100"/>
        <param name="use_tf_static" value="true"/>
        <param name="tf_prefix" value="$(arg tf_prefix)"/>
        <remap from="robot_description" to="robot_description_planar"/>
        <remap from="joint_states" to="/joint_states"/>
    </node>

    <arg name="params_file" default="$(find husky_panda_manipulation)/config/holonomic_params.yaml" if="$(arg holonomic)"/>
    <arg name="params_file" default="$(find husky_panda_manipulation)/config/nonholonomic_params.yaml" unless="$(arg holonomic)"/>

    <node name="husky_panda_control" type="husky_panda_control" pkg="husky_panda_manipulation" output="screen" launch-prefix="$(arg launch_prefix)">    
        <rosparam command="load" file="$(arg params_file)"/>
        <rosparam command="load" file="$(find husky_panda_manipulation)/config/filter_params.yaml"/>
        <param name="config_file" value="$(arg params_file)"/>
        <param name="robot_description_dynamics" value="$(arg robot_description_file)"/>
        <param name="holonomic" value="$(arg holonomic)"/>
    </node>

    <include file="$(find husky_panda_manipulation)/launch/common.launch">
        <arg name="plot" value="$(arg plot)"/>
        <arg name="enable_obstacle" value="$(arg enable_obstacle)"/>
        <arg name="experiment" value="$(arg experiment)"/>
        <arg name="experiment_id" default="$(arg experiment_id)"/>
    </include>
</launch>
